src/librawspeed/decoders/SrwDecoder.cpp

```diff
@@ -52,21 +52,15 @@ RawImage SrwDecoder::decodeRawInternal() {
 
   if (32769 == compression)
   {
-    bool bit_order = false;  // Default guess
-    auto msb_hint = hints.find("msb_override");
-    if (msb_hint != hints.end())
-      bit_order = ("true" == (msb_hint->second));
+    bool bit_order = hints.get("msb_override", false);
     this->decodeUncompressed(raw, bit_order ? BitOrder_Jpeg : BitOrder_Plain);
     return mRaw;
   }
 
   if (32770 == compression)
   {
     if (!raw->hasEntry ((TiffTag)40976)) {
-      bool bit_order = (bits == 12);  // Default guess
-      auto msb_hint = hints.find("msb_override");
-      if (msb_hint != hints.end())
-        bit_order = ("true" == (msb_hint->second));
```

Comment:
Well, turns out, you had a point there. I went back to see what happens if you put 'True' in there and found out that the code did in fact not work as I claimed, i.e. calling it with 'true' returned `false`. I changed one line of the patch to fix it: `iss >> std::boolalpha >> defaultValue;`. The `boolalpha` modifier was missing.

It turns out, there is not a single hint in the xml that depends on evaluating 'true' to `true`. There are only 5 'true' strings and all those hints are only checked for their presence. I think this should be improved by making it more obvious: remove the 'value' property from the xml of those hints that are true iff they are present.